From 1dfc515d7e0b4aac4e85bf9200b652a7d0d65707 Mon Sep 17 00:00:00 2001
From: Joseph Chen <chenjh@rock-chips.com>
Date: Sat, 4 Nov 2017 14:31:48 +0800
Subject: [PATCH] ARM: dts: rk3066a-rayeager: add vdd_logic regulator

The vdd_logic is a pwm regulator. Since '#pwm-cells = <2>', there
is not polarity invert support by pwm driver, so we have to add
property 'pwm-dutycycle-range = <100 0>' to support polarity invert
by pwm regulator driver itself.

Change-Id: Ie5d2cda67ce19dc792f96263836bab658d385681
Signed-off-by: Joseph Chen <chenjh@rock-chips.com>
---
 arch/arm/boot/dts/rk3066a-rayeager.dts | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/boot/dts/rk3066a-rayeager.dts b/arch/arm/boot/dts/rk3066a-rayeager.dts
index 20151570af8b..572eab49b9a4 100644
--- a/arch/arm/boot/dts/rk3066a-rayeager.dts
+++ b/arch/arm/boot/dts/rk3066a-rayeager.dts
@@ -59,6 +59,20 @@
 		pinctrl-0 = <&ir_int>;
 	};
 
+	vdd_logic: vdd-logic {
+		compatible = "pwm-regulator";
+
+		pwms = <&pwm3 0 25000>;
+		pwm-dutycycle-range = <100 0>;
+
+		regulator-name = "vdd_logic";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <925000>;
+		regulator-max-microvolt = <1400000>;
+		status = "okay";
+	};
+
 	keys: gpio-keys {
 		compatible = "gpio-keys";
 		#address-cells = <1>;
@@ -500,6 +514,10 @@
 	status = "okay";
 };
 
+&pwm3 {
+	status = "okay";
+};
+
 &saradc {
 	vref-supply = <&vcc_25>;
 	status = "okay";
-- 
2.35.3

